packages/charts/react-charting/src/components/DeclarativeChart/DeclarativeChart.tsx

```diff
@@ -32,6 +32,15 @@ export interface DeclarativeChartProps extends React.RefAttributes<HTMLDivElemen
   chartSchema: any;
 }
 
+const useColorMapping = () => {
+  const colorMapping = React.useRef(new Map<string, string>());
+  const colorIndex = React.useRef(0);
+  return {
+    colorMapping,
+    colorIndex,
```

Comment:
colorIndex variable is redundant. It can be inferred using colorMapping.size

Reply:
in colorMapping, the colors which are used for empty colorStrings are not stored. So, we will lose the count in those cases without colorIndex.